plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -113,7 +115,16 @@ protected final void commitNewTable(TableMetadata metadata)
         Table table = builder.build();
 
         PrincipalPrivileges privileges = owner.map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
-        metastore.createTable(table, privileges);
+        try {
+            metastore.createTable(table, privileges);
+        }
+        catch (SchemaNotFoundException
+               | TableAlreadyExistsException
+               | UnsupportedOperationException e) {
```

Comment:
Why UnsupportedOperationException?

Reply:
I was looking for `createTable` implementation and the exceptions thrown by them. `InMemoryThriftMetastore#createTable` throw `UnsupportedOperationException`.

But now I realized that `InMemoryThriftMetastore` is implemented for test cases. So `UnsupportedOperationException` check is needed.